lib/fog/vsphere/models/compute/volume.rb

```diff
@@ -49,7 +49,8 @@ def save
           requires :server_id, :size, :datastore
 
           if unit_number.nil?
-            used_unit_numbers = server.volumes.map { |volume| volume.unit_number }
+            # Vsphere maps unit_numbers 7 and greater to a higher SCSI ID since the pvscsi driver reserves SCSI ID 7
+            used_unit_numbers = server.volumes.map { |volume| volume.unit_number < 7 ? volume.unit_number : volume.unit_number - 1 }
```

Comment:
```used_unit_numbers``` refers to the unit_numbers that fog has used.  When fog goes to add a volume with unit_number 7 for example, the SCSI ID ends up being 8.

Hope that provides some clarification.